instance/models/openedx_instance.py

```diff
@@ -249,54 +243,107 @@ def delete(self, *args, **kwargs):
         self.deprovision_swift()
         super().delete(*args, **kwargs)
 
-    @property
-    def appserver_set(self):
-        """
-        Get the set of OpenEdxAppServers owned by this instance.
-        """
-        return self.ref.openedxappserver_set
-
-    def set_appserver_active(self, appserver_id):
+    def set_dns_records(self):
         """
-        Mark the AppServer with the given ID as the active one.
+        Create CNAME records for the domain names of this instance pointing to the load balancer.
         """
-        app_server = self.appserver_set.get(pk=appserver_id)  # Make sure the AppServer is owned by this instance
-        self.logger.info('Making %s active for instance %s...', app_server.name, self.name)
-        public_ip = app_server.server.public_ip
+        load_balancer_domain = self.load_balancing_server.domain.rstrip(".") + "."
 
         self.logger.info('Updating DNS: LMS at %s...', self.internal_lms_domain)
         lms_domain = tldextract(self.internal_lms_domain)
         gandi.set_dns_record(
             lms_domain.registered_domain,
-            type='A', name=lms_domain.subdomain, value=public_ip
+            type='CNAME', name=lms_domain.subdomain, value=load_balancer_domain
         )
 
         self.logger.info('Updating DNS: LMS preview at %s...', self.internal_lms_preview_domain)
         lms_preview_domain = tldextract(self.internal_lms_preview_domain)
         gandi.set_dns_record(
             lms_preview_domain.registered_domain,
-            type='CNAME', name=lms_preview_domain.subdomain, value=lms_domain.subdomain
+            type='CNAME', name=lms_preview_domain.subdomain, value=load_balancer_domain
         )
 
         self.logger.info('Updating DNS: Studio at %s...', self.internal_studio_domain)
         studio_domain = tldextract(self.internal_studio_domain)
         gandi.set_dns_record(
             studio_domain.registered_domain,
-            type='CNAME', name=studio_domain.subdomain, value=lms_domain.subdomain
+            type='CNAME', name=studio_domain.subdomain, value=load_balancer_domain
         )
 
+    def get_load_balancer_configuration(self):
```

Comment:
@smarnach Alright, I see - I was hoping the load balancer configuration could be more independent of Open edX, but since that isn't the case, leaving that method on the OpenEdXInstance object works for me. 

> It would be possible to move some of the current code from LoadBalancingServer to a new class LoadBalancedInstance. However, I don't see how this would improve the code. You would just have to look in one more place to understand how the parts are working together.

If the LoadBalancer takes a raw configuration from the instance, you will end up with logic common to all the Instance objects that can't easily be moved to the LoadBalancer object. For example, in your current get_load_balancer_configuration method, the primary page server logic would be generic to all types of instances imho; but if you move that logic to the LoadBalancer object, you would have to return a special value to signify that a given instance wants to use the primary page. It would be easier to understand and more maintainable imho to have this logic on a method on LoadBalancingInstanceMixin which the get_load_balancer_configuration method can call. It is an additional place to look at, but that would help to remember to explicitly differentiate between the generic parts and the ones specific to Open edX.